### ISSUE TITLE
[question] Change log level for a particular log4j logger?

### ISSUE THREAD

**Issue by erikgb** [confluentinc/cp-ansible#433]
**Describe the issue**
We want to configure more detailed logging, typically DEBUG level, for a log4j logger on the Kafka broker. Is that an implemented feature?

It seems like the log4j-properties template reference is hard-coded in https://github.com/confluentinc/cp-ansible/blob/c41b2325e6521a602aa684743d18a4e8dc90febe/roles/confluent.kafka_broker/tasks/main.yml#L160
Could it be an option to add a new role variable allowing the user to supply a custom template? I'll guess that adding more template variables would just create a complete mess....

Comments or suggestions?

**Comment by JumaX** [comment]
@erikgb  Thanks for the question.  It depends on the branch of CP-Ansible you are running, on the 5.5.x-post branches it is hardcoded to that template file.  In the 6.0.x-post branches we have made every property customizable, so you can set your own custom log4j template.  We even have a module to copy files to the hosts for you.

https://docs.confluent.io/current/installation/cp-ansible/ansible-configure.html#set-custom-component-properties

https://docs.confluent.io/current/installation/cp-ansible/ansible-configure.html#ansible-copy-file

**Comment by erikgb** [comment]
@JumaX Thanks, we are on the 6.0.0-post branch - so we should be ok then. I wasn't aware of you ansible-copy-file module, and I thought that the Ansible copy module didn't support template files, but I will give it a try.

**Comment by JumaX** [comment]
@erikgb To clarify, I haven't run this first hand, but theoretically this should work. :). Let's leave this thread open and if you hit an issue, please update us and we can investigate/try to provide a solution.

**Comment by erikgb** [comment]
@JumaX Thanks! I am testing a few things now, and will get back to you. I am struggeling to "beat" this task: https://github.com/confluentinc/cp-ansible/blob/37aa0dad70cdc3e4df13e57cedfe4a7f6b7f7abe/roles/confluent.kafka_broker/tasks/main.yml#L158

Is there a default log4j.properties location? If not, I cannot win - since the system property supplied to the jvm is using the same variable as the `dest`-property in the task referred to above: https://github.com/confluentinc/cp-ansible/blob/9c7954d5d32bb5d6aa5e84f9d9570a5ee6ba54bb/roles/confluent.kafka_broker/defaults/main.yml#L13

I can set `kafka_broker_custom_log4j` to `false`, and use `kafka_broker_custom_java_args` to set my custom log4j.properties. This might work, but seems a bit counter intuitive....

**Comment by domenicbove** [comment]
Ya @erikgb nice catch. There should be a conditional on the `Create Kafka Broker log4j Config`

Right now this would unnecessarily create our log4j file, but wont add it to the java args, and instead add yours:
```
kafka_broker_custom_log4j: false
kafka_broker_copy_files:
  - source_path: /path/to/custom_log4j.properties
     destination_path: /etc/kafka/custom_log4j.properties
kafka_broker_custom_java_args: -Dlog4j.configuration=file:/etc/kafka/custom_log4j.properties
```

**Comment by erikgb** [comment]
@domenicbove Exatcly what I am trying out now. It'll will do for now, but should be improved. 😄 I'll back to you.

**Comment by erikgb** [comment]
My "wingman", that can approve my merge request, just left for today, so I'll have to get back to you tomorrow. But I think the workaround will work.

**Comment by erikgb** [comment]
@domenicbove @JumaX I have now verified that your suggested solution works, with some minor quirks:

1. I have to set the property `kafka_broker_custom_log4j` to `false`, which is counterintuitive.
1. The `kafka_broker_copy_files` variable does not support template `source_path`s, which is kinda expected (from the name), but unfortunate for us. Our custom log4j-configuration started off as a copy of the template file within the `confluent.kafka_broker` role. And the default values of variables used in that template is inside the role, and thus not easy accessible (before applying the role). But I managed to work around this issue too. Currently our log4j-configuration is a static file. If we need templating, we will just add it - and render the template on the Ansible controller.

Thanks for all your help! Closing this issue now, and leaving up to you to decide if you want to create any other issue(s) based on this "question".
